elastic/kibana
# [Response Ops][Alerting] Only load maintenance windows when there are alerts during rule execution and caching loaded maintenance windows

Resolves https://github.com/elastic/kibana/issues/184324

## Summary

This PR moves the loading of maintenance windows further down in rule execution so maintenance windows are only loaded when a rule execution generates alerts. Also caches maintenance windows per space to reduce the number of requests.

## To Verify

1. Add some logging to x-pack/plugins/alerting/server/task_runner/maintenance_windows/maintenance_windows_service.ts to indicate when windows are being fetched and when they're returning from the cache.
2. Create and run some rules in different spaces with and without alerts to see that the maintenance windows are only loaded when there are alerts and that the windows are returned from the cache when the cache has not expired.
